packages/in_app_purchase/lib/src/in_app_purchase/app_store_connection.dart

```diff
@@ -208,12 +208,14 @@ class _TransactionObserver implements SKTransactionObserverWrapper {
         return wrapper.transactionState ==
             SKPaymentTransactionStateWrapper.restored;
       }).map((SKPaymentTransactionWrapper wrapper) => wrapper));
-      return;
     }
 
     String receiptData = await getReceiptData();
     purchaseUpdatedController
-        .add(transactions.map((SKPaymentTransactionWrapper transaction) {
+        .add(transactions.where((SKPaymentTransactionWrapper wrapper) {
+      return wrapper.transactionState !=
+          SKPaymentTransactionStateWrapper.restored;
+    }).map((SKPaymentTransactionWrapper transaction) {
```

Comment:
@LHLL and @cyanglaz 

> Make sure queryPastPurchases() won't block transaction updates. When the client apps calls queryPastPurchases() right after app launch, they should still receive transaction updates in the purchaseUpdatedStream for transactions with state != restored.

Why did you add this filter and only give transactions that are not restored? Seems like this has resulted in a different bug, see: https://github.com/flutter/flutter/issues/43957#issuecomment-811898757